src/library_webgl2.js

```diff
@@ -127,15 +127,22 @@ var LibraryWebGL2 = {
         return HEAPU16;
       case 0x1404 /* GL_INT */:
         return HEAP32;
+      case 0x1406 /* GL_FLOAT */:
+        return HEAPF32;
+      default:
+#if GL_ASSERTIONS
+        err('GL_INVALID_PARAMETER error in _heapObjectForWebGLType! Specified GL color format type=0x' + type.toString(16) + ' is not supported!');
+#endif
+        // Do not record a GL error, since the users of _heapObjectForWebGLType will call out to a WebGL function, so browser will record an error.
+        // Because of the ensuing error, we can return an arbitrary type from here. Choose to return the HEAPU32 type since most of the enumerant
+        // values are of this type: (that way Closure compiler might be able to fold these away)
       case 0x1405 /* GL_UNSIGNED_INT */:
       case 0x84FA /* GL_UNSIGNED_INT_24_8_WEBGL/GL_UNSIGNED_INT_24_8 */:
       case 0x8C3E /* GL_UNSIGNED_INT_5_9_9_9_REV */:
       case 0x8368 /* GL_UNSIGNED_INT_2_10_10_10_REV */:
       case 0x8C3B /* GL_UNSIGNED_INT_10F_11F_11F_REV */:
       case 0x84FA /* GL_UNSIGNED_INT_24_8 */:
         return HEAPU32;
```

Comment:
for code size, you could remove those cases, and in the default handler, in the GL_ASSERTIONS code, check that the value is one of those? Then the size is just on GL_ASSERTIONS builds, and doesn't depend on closure.

Reply:
Oh this code path actually just got superceded by #9938, so I'll refactor this to land only the test part, and land the micro optimize PR, that will fix the behavior as well. (#9938 does it like the way you describe)